Automattic/jetpack
# Publicize: Allow publishing a post as a Twitter thread.

# Project Goal

The goal of this project is to create a seamless method for fully featured posts to be transformed automatically into Twitter Threads, fitting into the Twitter format as naturally as possible. Along the way, it should provide useful feedback to the author, indicating how their post will be transformed, and giving them the opportunity to tweak their post content based on how they would like the thread to appear.

# Overviews

For an overview of the logic behind mapping post content to tweets, I recommend reading the [Content Overview](https://github.com/Automattic/jetpack/pull/16699#issuecomment-690009821).

If you're after technical information about how all of the parts work together, then you'll want to head over to the [Technical Overview](https://github.com/Automattic/jetpack/pull/16699#issuecomment-690010800).

If you want to learn how you can most effectively help with testing, and what you can look for to indicate incorrect behaviour that should be reported, check out the [Product Testing Overview](https://github.com/Automattic/jetpack/pull/16699#issuecomment-693176464).

Finally, if there's a feature missing that you feel should be included, please check the [Future Work Overview](https://github.com/Automattic/jetpack/issues/17149) first, as there are [alot](//core.trac.wordpress.org/ticket/50000) of things I'd love to include, but not everything will make it into the first release.

# Screenshots

<img width="400" alt="" src="https://user-images.githubusercontent.com/352291/90095597-7a898880-dd74-11ea-952b-e090d0bc8ed5.png"> <img width="400" alt="" src="https://user-images.githubusercontent.com/352291/90365128-bab87600-e0a8-11ea-8a46-8a404ec763ca.png">
 <img width="400" alt="" src="https://user-images.githubusercontent.com/352291/90364190-5b0d9b00-e0a7-11ea-803f-a30a904d765c.png"> <img width="400" alt="" src="https://user-images.githubusercontent.com/352291/91515253-1e099a00-e92c-11ea-91a5-89fa8bc45019.png"> <img width="400" alt="" src="https://user-images.githubusercontent.com/352291/91515250-1c3fd680-e92c-11ea-959a-3b3215d9ed3c.png"> <img width="400" alt="" src="https://user-images.githubusercontent.com/352291/91515244-177b2280-e92c-11ea-9210-0927832a8ee2.png">

# Housekeeping

#### Changes proposed in this Pull Request:

* Adds a new feature: Publishing Twitter threads from within the block editor, via Publicize.

#### Jetpack product discussion

p1HpG7-9US-p2

#### Does this pull request change what data or activity we track or use?

No change, it only uses existing data (eg, Keyring for Twitter connection, post content for generating thread)

#### Testing instructions:

##### Setup

If you need to test parser changes on your sandbox:

- Apply the relevant diff to your sandbox.
- Ensure you've added `add_filter( 'publicize_force_synchronous', '__return_true' );` on your sandbox.
- Define `JETPACK__SANDBOX_DOMAIN` to your sandbox domain in the `wp-config.php` file on your Jetpack site.

##### Testing

Please see the [Product Testing Overview](https://github.com/Automattic/jetpack/pull/16699#issuecomment-693176464) details on testing.

#### Proposed changelog entry for your changes:

* Publicize: Add the ability to publish the entire content of posts to Twitter as threads.
